cmd/main.go

```diff
@@ -197,31 +215,48 @@ func preRun(cmd *cobra.Command, args []string) {
 			log.Fatal().Msg(err.Error())
 		}
 	}()
+
+	return nil
 }
 
-func postRun(cmd *cobra.Command, args []string) {
+func postRun(cmd *cobra.Command, args []string) error {
 	channels.WaitGroup.Wait()
 
 	cfg := config.LoadConfig("2ms", Version)
 
 	// -------------------------------------
 	// Show Report
 	if report.TotalItemsScanned > 0 {
-		report.ShowReport(stdoutFormatVar, cfg)
+		if err := report.ShowReport(stdoutFormatVar, cfg); err != nil {
+			return err
+		}
+
 		if len(reportPathVar) > 0 {
 			err := report.WriteFile(reportPathVar, cfg)
 			if err != nil {
-				log.Error().Msgf("Failed to create report file with error: %s", err)
+				return fmt.Errorf("failed to create report file with error: %s", err)
 			}
 		}
 	} else {
-		log.Error().Msg("Scan completed with empty content")
-		os.Exit(0)
+		fmt.Println("Scan completed with empty content")
 	}
 
-	if report.TotalSecretsFound > 0 {
-		os.Exit(1)
-	} else {
-		os.Exit(0)
+	return nil
+}
+
+// InitShouldIgnoreArg initializes what kind of errors should be used on exit codes
+func InitShouldIgnoreArg(arg string) error {
+	validArgs := []string{"none", "all", "results", "errors"}
+	for _, validArg := range validArgs {
+		if strings.EqualFold(validArg, arg) {
+			ignoreOnExitVar = strings.ToLower(arg)
+			return nil
+		}
 	}
+	return fmt.Errorf("unknown argument for --ignore-on-exit: %s\nvalid arguments:\n  %s", arg, strings.Join(validArgs, "\n  "))
+}
+
+// ShowError returns true if should show error, otherwise returns false
+func ShowError(kind string) bool {
+	return strings.EqualFold(ignoreOnExitVar, "none") || (!strings.EqualFold(ignoreOnExitVar, "all") && !strings.EqualFold(ignoreOnExitVar, kind))
 }
```

Comment:
Remove the comment, and maybe split the long condition to make it readable

Reply:
done